react/features/post-meeting/middleware.js

```diff
@@ -0,0 +1,23 @@
+// @flow
+
+import { CONFERENCE_WILL_LEAVE } from '../base/conference';
+import { setActiveModalId } from '../base/modal';
+import { MiddlewareRegistry } from '../base/redux';
+
+import { POST_MEETING_MODAL_ID } from './constants';
+
+/**
+ * Implements the middleware of the post-meeting feature.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(({ dispatch }) => next => action => {
+    switch (action.type) {
+    case CONFERENCE_WILL_LEAVE:
+        dispatch(setActiveModalId(POST_MEETING_MODAL_ID));
```

Comment:
I don't think we want to show it if a new conference is being joined while we disconnect from this one. I suspect appNavigate may make it flicker by showing it first, then quickly hiding it. We should avoid that.

In more general terms, doing this for every conference termination is highly annoying. Not a single app similar to ours does it. I think we should use some form of randomized percentage. Let's check with product on this please.